docs/Custom-brimcap-load-Config.md

```diff
@@ -0,0 +1,433 @@
+        pcap_cnt: uint64,
+        tx_id: uint64,
+        icmp_code: uint64,
+        icmp_type: uint64,
+        tunnel: {
+          src_ip: ip,
+          src_port: port=(uint16),
+          dest_ip: ip,
+          dest_port: port=(uint16),
+          proto: bstring,
+          depth: uint64
+        },
+        community_id: bstring
+      }
+      filter event_type=alert | put . = shape(alert) | rename ts=timestamp
+```
+
+A full description of all that's possible with shaping configurations is beyond
+the scope of this article. However, this configuration is quite simple and can
+be described in brief.
+
+1. The `type alert` defines the names, [data types](https://github.com/brimdata/zed/blob/main/docs/formats/zson.md#33-primitive-values),
+and hierarchical locations of expected fields in the input records. Here we've
+defined a single "wide" shape for _all_ alerts we've known Suricata to
+generate, which is convenient because it allows Brim to easily display them in
+columnar format.
+
+2. The `filter event_type=alert` trims the processed EVE events to only alerts.
+If you want to let through more Suricata data besides just alerts, you could
+remove this part of the pipeline. If so, you'll likely want to explore the
+additional data and create shapers to apply proper data types to them, since
+this will be a prerequisite for doing certain Zed queries with the data (e.g.
+a successful CIDR match requires IP addresses to be stored as `ip` type, not
+the string type in which they'd appear in unshaped JSON).
+
+3. The `put . = shape(alert)` applies the shape of the `alert` type to each
+input record. With what's shown here, additional fields that appear beyond
+those specified in the shaper (e.g. as the result of new Suricata features or
+your own customizations) will still be let through this pipeline and stored in
+the Zed Lake. If this is undesirable, add `| put . = crop(alert)` downstream of
+the first `put`, which will trim these additional fields.
+
+4. The `rename ts=timestamp` changes the name of Suricata's `timestamp` field
+to match the `ts` one used by Zeek, which allows the data from both sources to
+be more seamlessly presented together in Brim.
+
+# Custom NetFlow Analyzer
+
+## Background
+
+Building on what we just learned about Brimcap analyzers, we can extend the
+concepts to pcap processing tools other than Zeek or Suricata. In this case
+we'll use NetFlow records generated using the open source
+[nfdump](https://github.com/phaag/nfdump) toolset.
+
+* **Note**: While the example shows how shaped NetFlow records can be loaded
+via Brimcap, the **Packets** button in Brim is not currently wired to extract
+flows from a pcap via the 5-tuple and time details in NetFlow records. If this
+is functionality you're interested in pursuing, please
+[open an issue](https://github.com/brimdata/brimcap/wiki/Troubleshooting#opening-an-issue)
+or come talk to us on [public Slack](https://www.brimsecurity.com/join-slack/).
```

Comment:
I think this is worth emphasizing. Whether NetFlow represents great pcap summary logs is perhaps open for debate. But FWIW, the combo of nfdump's fields `ts`, `td`, `sa`, `da`, `sp`, `dp`, and `pr` give us the info we'd need to form a `brimcap search` command line that would extract a flow. This combination of fields could be communicated to the app (perhaps via the YAML?) so they could be used when extracting the flows.